src/System.ServiceModel.Primitives/tests/ServiceModel/ChannelFactoryTest.cs

```diff
@@ -200,4 +201,28 @@ public static void CreateChannel_Of_Typed_Proxy_Using_BasicHttpBinding()
             }
         }
     }
+
+    [Fact]
+    public static void ChannelFactory_Async_Open_Close() {
+        ChannelFactory<IRequestChannel> factory = null;
+        try {
+            BasicHttpBinding binding = new BasicHttpBinding();
+
+            // Create the channel factory
+            factory = new ChannelFactory<IRequestChannel>(binding, new EndpointAddress(FakeAddress.HttpAddress));
+
+            Assert.Equal(CommunicationState.Created, factory.State);
+
+            Task.Factory.FromAsync(factory.BeginOpen(null, null), factory.EndOpen).GetAwaiter().GetResult();
+
+            Assert.Equal(CommunicationState.Opened, factory.State);
+
+        }
+        finally {
+            Assert.NotNull(factory);
+            
+            Task.Factory.FromAsync(factory.BeginClose(null, null), factory.EndClose).GetAwaiter().GetResult();
+            Assert.Equal(CommunicationState.Closed, factory.State);
```

Comment:
Consider moving the Begin/EndClose into the try block, then Abort in the finally block if CommunicationState != Closed.